core/project.cpp

```diff
@@ -62,19 +62,22 @@ ProjectStatus::Status ProjectStatus::projectStatus( const std::shared_ptr<Projec
     return ProjectStatus::NoVersion;
   }
 
-  //
-  // TODO: this check for local modifications should be revisited
-  //
-
   // Something has locally changed after last sync with server
   QString metadataFilePath = project->local->projectDir + "/" + MerginApi::sMetadataFile;
   QDateTime lastModified = CoreUtils::getLastModifiedFileDateTime( project->local->projectDir );
-  QDateTime lastSync = QFileInfo( metadataFilePath ).lastModified();
+  QDateTime lastSync = QFileInfo( metadataFilePath ).lastModified().toUTC();
   MerginProjectMetadata meta = MerginProjectMetadata::fromCachedJson( metadataFilePath );
   int filesCount = CoreUtils::getProjectFilesCount( project->local->projectDir );
   if ( lastSync < lastModified || meta.files.count() != filesCount )
   {
-    return ProjectStatus::Modified;
+    // When GPKG is opened, its header is updated and therefore lastModified timestamp is updated as well.
+    // Double check if there is really something to upload
+    QList<MerginFile> localFiles = MerginApi::getLocalProjectFiles( project->local->projectDir + "/" );
+    MerginProjectMetadata serverProject = MerginProjectMetadata::fromCachedJson( metadataFilePath );
+    ProjectDiff diff = MerginApi::compareProjectFiles( serverProject.files, serverProject.files, localFiles, project->local->projectDir );
+
+    if ( !diff.localAdded.isEmpty() || !diff.localDeleted.isEmpty() || !diff.localUpdated.isEmpty() )
```

Comment:
in else branch we do not return anything?

Reply:
Its not needed. The branch from line 71 covers only local changes case - we can quickly compare timestamps and check files diff afterwards only when needed. If there are no local changes in diff, we need to go further and check for server changes.

On the other hand, we need to check diff only when timestamp of gpkg file is the one, which fulfil condition on line 71 (gpkg file seems to be update after last sync, but actually gpkg data are not changed - thats the reported bug).

Reply:
bool function must return something all the time?

Reply:
Indeed, it has to. However, in this case, we have several if conditions (not even if else or else) which are evaluated one of the other - if one condition is not fulfilled, app continues with others. If none of them are fulfilled, function returns "default" `up to date` status.

This fix only extend current condition for `modified` status to be sure that correct status is returned.